### DISCUSSION TITLE
ETABSAdapter Component Error

### DISCUSSION THREAD

**Discussion by gizusabiteofyourpieau** [BHoM/BHoM#1416] [Q&A]
Hello 
I'm trying to set up an ETABs adaptor and I get the following error on the component:

> 1. This component failed to run properly.
>  Error: Failed to read from an IPC Port: The pipe has been ended.
>
>  Occured in  BH.Adapter.ETABS.ETABSAdapter..ctor(String filePath, EtabsSettings etabsSetting, Boolean active)
>    called from BH.UI.Base.Caller.Run(List`1 inputs)
>  Are you sure you have the correct type of inputs? Check their description for more details.

What does this mean?

This is the full graph:

![image](https://user-images.githubusercontent.com/39002239/179443165-c831f6e4-3f98-4900-a7a9-ae97fa8cdb4d.png)

again with the error:

![image](https://user-images.githubusercontent.com/39002239/179443117-035418f7-75bf-4e18-94a0-76c612f9eacf.png)

When I set the toggle to TRUE, ETABs starts and displays the splash screen but before getting to the main window, closes and then I see this error as above.
ETABS works fine on my system with licenses working properly. I am running ETABS 20.0.1

Kind Regards,
Jono

**Comment by alelom** [comment]
Hey @gizusabiteofyourpieau, thanks for posting. I read that you are using ETABS 20.0.1. Unfortunately, the toolkit still does not support that, see the [supported versions](https://github.com/BHoM/ETABS_Toolkit#known-versions-of-software-supported). Feel free to open up an issue on that toolkit to request this feature (support of V20), and, if you can, please contribute to the code to make it happen :)

**Reply by gizusabiteofyourpieau** [reply]
Hi @alelom. I tried a few more times and it now works fine with V20 with no errors. Does that make sense considering it's not supported?
It would be good to understand what this error is and what it means

**Reply by alelom** [reply]
Sorry, this error is specific to ETABS, not BHoM, so it is a bit difficult to replicate.

The [supported versions](https://github.com/BHoM/ETABS_Toolkit#known-versions-of-software-supported) indicate versions that are sure to be working because we could not test them (we do not own every license for every version of ETABS); it does not automatically exclude that other versions may work. Versions older that the oldest supported version are unlikely to work, while versions newer that the newest supported version may work. We may decide to create a public procedure for others, like you, to report that a version is fully tested and working. Feedback like yours is valuable.

Can we consider this issue closed?

**Reply by gizusabiteofyourpieau** [reply]
I'm not getting the error anymore so we can consider this issue closed, thanks
